content/docs/extensions/online_advisor.md

```diff
@@ -0,0 +1,176 @@
+---
+title: The online_advisor extension
+subtitle: Get index, statistics, and prepared statement recommendations based on your query workload
+enableTableOfContents: true
+updatedOn: '2025-08-15T00:26:53.000Z'
+---
+
+The `online_advisor` extension recommends **indexes**, **extended statistics**, and **prepared statements** based on your actual query workload. It uses the same executor hook mechanism as [`auto_explain`](https://www.postgresql.org/docs/current/auto-explain.html) to collect and analyze execution data.
+
+<CTA />
+
+## What it does
+
+- Suggests **indexes** when queries filter many rows
+- Suggests **extended statistics** when the planner's row estimates are far off from actuals
+- Identifies queries that could benefit from **prepared statements** when planning time is high
+
+<Admonition type="note">
+`online_advisor` only makes recommendations. It does not create indexes or statistics for you.
+</Admonition>
+
+## Requirements
+
+- PostgreSQL **14–17**
+- For Postgres 14-16 only, `online_advisor` must be added to [`shared_preload_libraries`](/docs/extensions/pg-extensions#extensions-with-preloaded-libraries) before you can install it via `CREATE EXTENSION`.
+- Create the extension in every database you want to inspect
+- Activate it by calling any provided function (for example, `get_executor_stats()`)
+
+### Version availability
+
+Please refer to the [Supported Postgres extensions](/docs/extensions/pg-extensions) page for the latest supported version of `online_advisor` on Neon.
+
+## Enable the online_advisor extension
+
+<Admonition type="important">
+Before you can create the `online_advisor` extension **on Neon Postgres 14-16**, you must enable the `online_advisor` preloaded library. See [Extensions with preloaded libraries](/docs/extensions/pg-extensions#extensions-with-preloaded-libraries) for instructions. This step is not required on Postgres 17 projects.
+</Admonition>
+
+You can create the extension in each target database using `CREATE EXTENSION`:
+
+```sql
+CREATE EXTENSION online_advisor;
+```
+
+For information about using the Neon SQL Editor, see [Query with Neon's SQL Editor](/docs/get-started/query-with-neon-sql-editor). For information about using the `psql` client with Neon, see [Connect with psql](/docs/connect/query-with-psql-editor).
+
+## Start collecting recommendations
+
+1. (Optional) Before activation, set limits on how many proposals to track:
```

Comment:
Remove this step. You can't do this one Neon. Users cannot modify system settings.